source/browser.ts

```diff
@@ -657,6 +665,7 @@ function isPreferencesOpen(isNewDesign: boolean): boolean {
 function closePreferences(isNewDesign: boolean): void {
 	if (isNewDesign) {
 		const closeButton = document.querySelector<HTMLElement>('[aria-label=Preferences] [aria-label=Close]')!;
+		document.documentElement.classList.remove('hide-preferences-window');
 		return closeButton.click();
```

Comment:
You should first click the close button and only after that remove the utility class.

Just reverse the order of calls.
```suggestion
		closeButton.click();
		return document.documentElement.classList.remove('hide-preferences-window');
```

Reply:
That still didn't work perfectly, I added the `MutationObserver` instead.